seccomp_gen/pkg/tracing/tracer.go

```diff
@@ -0,0 +1,96 @@
+package tracing
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+)
+
+/*
+ * /falco/rules.yaml - file that falco uses to get the rules 
+ * /falco/data.json - traced syscalls list are stored here
+ * /falco/logs.txt - falco logs are stored here (this includes the logs from the `formatter` binary)
+*/
+
+type TracingConfiguration struct {
+  PodName string `json:"podName"`
+  ContainerName string `json:"containerName"`
+  PodLabel map[string]string `json:"podLabel"`
+}
+
+type Tracer struct { 
+  falcoProcess *os.Process
+  Config TracingConfiguration
+}
+
+func NewTracer() (Tracer, error) {
+  tracer := Tracer{}
+  return tracer, nil
+}
+
+
+func (t *Tracer) UpdateConfig(config TracingConfiguration) error {
+  t.Config = config
+  rule, err := CreateFalcoRule(config)
+  if err != nil {
+      return err
+  }
+  // Write the rule to /falco/rules.yaml
+  err = os.WriteFile("/falco/rules.yaml", rule, 0644)
+  if err != nil {
+    return err
+  }
+  return nil
+}
+
+// Start Falco process and update the struct with the falco process
+func (t *Tracer) Start() error {
+  khost := os.Getenv("KUBERNETES_SERVICE_HOST")
+  if khost == "" {
+    return fmt.Errorf("KUBERNETES_SERVICE_HOST not set")
+  }
+  falcoCommand := exec.Command("/usr/bin/falco",
+    // to get support for all available syscalls
+    "-A",
+    // unbuffered makes sure that we don't lose any syscalls
+    "-U",
+    "-r", "/falco/rules.yaml", 
+    "-k", "https://"+khost,
+    "-K", "/var/run/secrets/kubernetes.io/serviceaccount/token",
+    // TODO: What if crio is not used?
```

Comment:
+1 this needs to be compatible with containerd too. Not sure if k8s exposes somewhere the runtime socket. But this should be a falco common issue, maybe just ask the question in an issue or in the community chat

Reply:
I don't think we have to pass this option. It says in the documentation that it uses `libs` by default. And unless we find an issue with this default approach of using `libs`, I don't think we need to pass this argument.